File: app/models/job.rb
class Job
  include Mongoid::Document
  include Mongoid::Timestamps

  belongs_to :user
  belongs_to :datasource

  field :name, type: String
  field :filter, type: String
  field :types, type: String
  field :mongo_query, type: String
  field :public, type: Boolean

  field :started, type: DateTime
  field :finished, type: DateTime
  field :status, type: String, default: "WAITING"

end

File: frontend/app/models/job.js
import { computed } from '@ember/object';
import DS from 'ember-data';

const { attr } = DS;

export default DS.Model.extend({
  user:  DS.belongsTo('user'),
  datasource: DS.belongsTo('datasource'),

  name: attr('string'),
  filter: attr(),
  mongo_query: attr(),
  public: attr(),
  description: attr(),

  started: attr('dates'),
  finished: attr('dates'),
  status: attr('status'),
  created_at: attr('dates'),
  updated_at: attr('dates'),

  idTruncated: computed('id', function() {
    return this.get('id').slice(-6);
  }),
  hasCompleted: computed('status', function() {
    return this.get('status').message === 'COMPLETED';
  }),
  isRunning: computed('status', function() {
    return this.get('status').message === 'RUNNING';
  }),
  noResult: computed('status', function() {
    return this.get('status').message === 'EMPTY';
  }),
  hasError: computed('status', function() {
    return this.get('status').message === 'ERROR';
  }),
});
